crates/matrix-sdk/src/event_cache/linked_chunk.rs

```diff
@@ -158,18 +236,25 @@ impl<const CAP: usize, Item, Gap> LinkedChunk<CAP, Item, Gap> {
                     if item_index == current_items_length {
                         chunk
                             // Push the new items.
-                            .push_items(items, &chunk_identifier_generator)
+                            .push_items(items, &chunk_identifier_generator, listener.as_ref())
                     }
                     // Insert inside the current items.
                     else {
+                        listener.truncate_items(chunk_identifier, item_index).unwrap();
```

Comment:
For API symmetry with the rest, could we have some helper method that does the `split_off` + `listener.truncate_items` at the same time? (Since it's repeated twice at least in the code below.)

Reply:
There is no more repetition as far as I know.